tests/test_google_verifier.py

```diff
@@ -64,3 +67,47 @@ def test_google_verify_product():
             ):
                 with pytest.raises(errors.GoogleError):
                     verifier.verify("test-token", "test-product")
+
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), 'data')
+
+
+def datafile(filename):
+    return os.path.join(DATA_DIR, filename)
+
+
+def test_broken_receipt():
+    with patch.object(
+        googleplay.GooglePlayVerifier, "_authorize", return_value=None
+    ):
+        verifier = GooglePlayVerifier('bundle_id', 'private_key_path')
+
+        auth = HttpMock(datafile("androidpublisher.json"), headers={'status': 200})
+
+        request_mock_builder = RequestMockBuilder(
+            {
+                'androidpublisher.purchases.products.get': (
+                    httplib2.Response({'status': 400, "reason": b"Bad request"}),
+                    b'{"reason": "Bad request"}',
+                )
+            }
+        )
+        build_mock_result = googleplay.build("androidpublisher", "v3", http=auth, requestBuilder=request_mock_builder)
+
+        with patch.object(googleplay, "build", return_value=build_mock_result):
+            with pytest.raises(errors.GoogleError, match="Bad request"):
+                verifier.verify('broken_purchase_token', 'product_scu')
+
+        request_mock_builder = RequestMockBuilder(
```

Comment:
why duplicate previously defined `request_mock_builder ` ?

Reply:
Sry, time pressure. I will split test in two parts